numpy/lib/function_base.py

```diff
@@ -1019,9 +1024,14 @@ def gradient(f, *varargs, **kwargs):
     elif np.issubdtype(otype, np.inexact):
         pass
     else:
-        # all other types convert to floating point
+        # All other types convert to floating point.
+        # First check if f is a numpy integer type; if so, convert f to float64
+        # to avoid modular arithmetic when computing the changes in f.
+        if np.issubdtype(otype, np.integer):
+            f = f.astype(np.float64)
```

Comment:
For consistency with the line below, can we make this `np.double`?

Reply:
Even more consistent with usage throughout the file is to change `double` to `float64`, so I'll do that.

Reply:
Done.